pkg/workload/tpcc/payment.go

```diff
@@ -140,13 +143,71 @@ func createPayment(ctx context.Context, config *tpcc, mcp *workload.MultiConnPoo
 		VALUES ($1, $2, $3, $4, $5, $6, $7, $8)`,
 	)
 
+	if p.config.isLongDurationWorkload {
+		p.resetWarehouse = p.sr.Define(`
+		UPDATE warehouse
+		SET w_ytd = $1
+		WHERE w_id >= $2 AND w_id < $3`,
+		)
+
+		p.resetDistrict = p.sr.Define(`
+		UPDATE district
+		SET d_ytd = $1
+		WHERE d_w_id >= $2 AND d_w_id < $3`,
+		)
+
+		// Starting the background goroutine which will reset the w_ytd values periodically in warehouseWytdResetPeriod
+		go p.startResetValueWorker(ctx)
+	}
+
 	if err := p.sr.Init(ctx, "payment", mcp); err != nil {
 		return nil, err
 	}
 
 	return p, nil
 }
 
+func (p *payment) startResetValueWorker(ctx context.Context) {
+	ticker := time.NewTicker(warehouseWytdResetPeriod)
+	defer ticker.Stop()
+
+	p.config.resetTableWg.Add(1)
+	defer p.config.resetTableWg.Done()
+
+	for {
+		select {
+		case <-ctx.Done():
+			return
+		case <-ticker.C:
+
+			// Creating batches of maxRowsToUpdateTxn to avoid long-running txns
+			for startRange := 0; startRange <= p.config.warehouses; startRange += maxRowsToUpdateTxn {
```

Comment:
If warehouses is 0-based, then `startRange` should be exclusive of `p.config.warehouses`; i.e., otherwise, it's an empty range.

Reply:
Thanks for pointing out !